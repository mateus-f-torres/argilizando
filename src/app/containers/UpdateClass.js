import { connect } from 'react-redux';
import { displayClass, showClassSelection } from '../actions';
import { doneClass, showMenu } from '../actions';
import PlayerClass from '../components/PlayerClass.jsx';

const mapStateToProps = (state) => {
  return {
    selected: state.gameClass.show,
    chosen: state.gameClass.chosen
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getClass: (choice) => {
      dispatch(displayClass(choice))
    },
    noClass: () => {
      dispatch(showClassSelection())
    },
    lockClass: () => {
      dispatch(doneClass())
      dispatch(showMenu())
    }
  }
}

const UpdateClass = connect(
  mapStateToProps,
  mapDispatchToProps
)(PlayerClass);

export default UpdateClass;
