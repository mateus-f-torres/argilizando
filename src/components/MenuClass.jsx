//@flow
import * as React from 'react';
import {Link, Route, Switch} from 'react-router-dom';
import CardClass from './CardClass.jsx';
import CLASSES from '../data/classes.js';

import 'Styles/MenuClass.scss';

type Props = {
  getClass: (string) => void,
  lockClass: ({}) => void,
  chosen: {},
  match: {
    url: string,
  },
};

class MenuClass extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
    this.getClass = this.getClass.bind(this);
    this.lockClass = this.lockClass.bind(this);
  }

  getClass(e) {
    this.props.getClass(e.target.name);
  }

  lockClass() {
    this.props.lockClass(this.props.chosen);
  }

  render() {
    return (
      <section id="CLASS">
        <Switch>

          <Route exact path="/class" render={()=>
            <ul className="selection">
              <div id="title-box">
                <h2>Player Class</h2>
                <div className="horizontal-line no-bottom" />
              </div>
              {
                CLASSES.map((category, i) => {
                  let [type, options] = [...category];

                  let odd = type === 'Defenders' || type === 'Controllers';
                  let two = type === 'Leaders';

                  return (
                    <div key={i}>
                      <h3>{type}</h3>
                      <div className="horizontal-line no-bottom"/>
                      <ul
                        id={two ? 'two-cell' : null}
                        className={odd ? 'odd-number' : null}>
                        {
                          options.map((item) => (
                            <li key={item.id}>
                              <Link to={this.props.match.url + '/' + item.id}>
                                <button name={item.id} onClick={this.getClass}
                                  className={
                                    item.id === 'wizard'
                                    || item.id === 'paladin'
                                      ? 'double-size-btn'
                                      : null}>
                                  {item.id}
                                </button>
                              </Link>
                            </li>
                          ))
                        }
                      </ul>
                    </div>
                  );
                })
              }
            </ul>
          }/>

          <Route path={this.props.match.url + '/'}
            render={()=>
              <CardClass lockClass={this.lockClass} {...this.props.chosen}/>
            }/>
        </Switch>
      </section>
    );
  }
}

export default MenuClass;

