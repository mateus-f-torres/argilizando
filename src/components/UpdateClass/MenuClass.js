//@flow
import * as React from 'react';
import {Link, Route, Switch} from 'react-router-dom';
import CardClass from './CardClass.js';
import CLASSES from 'DATA/classes.js';
import type {_Class} from 'Types/props.js';

type Props = {
  getClass: (string) => void,
  lockClass: ({}) => void,
  _class: _Class,
  match: {
    url: string,
  },
};

class MenuClass extends React.Component<Props> {
  getClass = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.getClass(e.currentTarget.name);
  }

  lockClass = () => {
    this.props.lockClass(this.props._class);
  }

  render() {
    return (
      <section id="CLASS">
        <Switch>

          <Route exact path="/class" render={()=>
            <ul className="selection">
              <div id="title-box">
                <h2>Player Class</h2>
                <div className="horizontal-line no-bottom" />
              </div>
              {
                CLASSES.map((category, i) => {
                  let [type, options] = [...category];

                  let odd = type === 'Defenders' || type === 'Controllers';
                  let two = type === 'Leaders';

                  return (
                    <div key={i}>
                      <h3>{type}</h3>
                      <div className="horizontal-line no-bottom"/>
                      <ul
                        id={two ? 'two-cell' : null}
                        className={odd ? 'odd-number' : null}>
                        {
                          options.map((item) => (
                            <li key={item.id}>
                              <Link to={this.props.match.url + '/' + item.id}>
                                <button name={item.id} onClick={this.getClass}
                                  className={
                                    item.id === 'wizard'
                                    || item.id === 'paladin'
                                      ? 'double-size-btn'
                                      : null}>
                                  {item.id}
                                </button>
                              </Link>
                            </li>
                          ))
                        }
                      </ul>
                    </div>
                  );
                })
              }
            </ul>
          }/>

          <Route path={this.props.match.url + '/:classID'}
            render={()=>
              <CardClass lockClass={this.lockClass} {...this.props._class}/>
            }/>
        </Switch>
      </section>
    );
  }
}

export default MenuClass;

