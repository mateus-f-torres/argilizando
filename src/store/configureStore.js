import {createStore} from 'redux';
import rootReducer from '../reducers/index.js';

export default function configureStore() {
  return createStore(rootReducer);
}
