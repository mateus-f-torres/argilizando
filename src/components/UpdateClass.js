//@flow
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayClass} from 'Store/classes/actions';
import {lockClass} from 'Store/done/actions';
import MenuClass from 'Components/UpdateClass/MenuClass.js';
import type {State} from 'Types/state.js';

const mapStateToProps = (state: State) => {
  return {
    _class: state._class,
  };
};

const mapDispatchToProps = (dispatch: *) => {
  return {
    getClass: (choice) => {
      dispatch(displayClass(choice));
    },
    lockClass: (gameClass) => {
      dispatch(lockClass(gameClass));
    },
  };
};

const UpdateClass = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(MenuClass));

export default UpdateClass;
