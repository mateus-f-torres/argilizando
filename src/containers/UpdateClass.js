//@flow
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {displayClass, lockClass} from '../actions';
import MenuClass from '../components/MenuClass.jsx';
import type {State} from '../types/state.js';

const mapStateToProps = (state: State) => {
  return {
    _class: state._class,
  };
};

const mapDispatchToProps = (dispatch: *) => {
  return {
    getClass: (choice) => {
      dispatch(displayClass(choice));
    },
    lockClass: (gameClass) => {
      dispatch(lockClass(gameClass));
    },
  };
};

const UpdateClass = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(MenuClass));

export default UpdateClass;
