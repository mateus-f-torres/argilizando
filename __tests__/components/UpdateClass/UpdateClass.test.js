import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';

import configureStore from 'redux-mock-store';
import {mount} from 'enzyme';

import UpdateClass from 'Components/UpdateClass.js';
import MenuClass from 'Components/UpdateClass/MenuClass.js';

const mockState = {
  _class: {},
};

const mockStore = configureStore();
const store = mockStore(mockState);

describe('UpdateClass', () => {
  beforeEach(() => {
    store.clearActions();
  });

  const container = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{pathname: '/class', key: 'testKey'}]}>
        <UpdateClass />
      </MemoryRouter>
    </Provider>
  );

  // TODO check lockClass dispatch

  describe('Unit Test', () => {
    test('render MenuClass component', () => {
      expect(container.containsMatchingElement(<MenuClass />)).toBe(true);
    });

    test('receive store._class', () => {
      const component = container.find('MenuClass');

      expect(component.prop('_class')).toEqual(mockState._class);
    });

    test('dispatch getClass', () => {
      const component = container.find('MenuClass');

      const barbarianBtn =
        component.find('button').filter('[name="barbarian"]');

      const expectedActions = [
        {
          'type': 'DISPLAY_CHOSEN_CLASS',
          'payload': 'barbarian',
        },
      ];

      barbarianBtn.simulate('click');
      expect(store.getActions()).toEqual(expectedActions);
    });

    test('dispatch lockClass', () => {});
  });

  describe('Snapshot Test', () => {
    expect(container).toMatchSnapshot();
  });
});
