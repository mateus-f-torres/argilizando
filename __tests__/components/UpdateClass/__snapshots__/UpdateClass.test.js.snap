// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/class",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/class",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/class",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(MenuClass))>
        <Route>
          <Connect(MenuClass)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/class",
                    "search": "",
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/class",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/class",
                "search": "",
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <MenuClass
              _class={Object {}}
              getClass={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/class",
                      "search": "",
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/class",
                    "search": "",
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/class",
                  "search": "",
                }
              }
              lockClass={[Function]}
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <section
                id="CLASS"
              >
                <Switch>
                  <Route
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/class",
                        "url": "/class",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/class",
                        "search": "",
                      }
                    }
                    path="/class"
                    render={[Function]}
                  >
                    <ul
                      className="selection"
                    >
                      <div
                        id="title-box"
                      >
                        <h2>
                          Player Class
                        </h2>
                        <div
                          className="horizontal-line no-bottom"
                        />
                      </div>
                      <div
                        key="0"
                      >
                        <h3>
                          Defenders
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul
                          className="odd-number"
                          id={null}
                        >
                          <li
                            key="barbarian"
                          >
                            <Link
                              replace={false}
                              to="//barbarian"
                            >
                              <a
                                href="//barbarian"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="barbarian"
                                  onClick={[Function]}
                                >
                                  barbarian
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="fighter"
                          >
                            <Link
                              replace={false}
                              to="//fighter"
                            >
                              <a
                                href="//fighter"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="fighter"
                                  onClick={[Function]}
                                >
                                  fighter
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="paladin"
                          >
                            <Link
                              replace={false}
                              to="//paladin"
                            >
                              <a
                                href="//paladin"
                                onClick={[Function]}
                              >
                                <button
                                  className="double-size-btn"
                                  name="paladin"
                                  onClick={[Function]}
                                >
                                  paladin
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="1"
                      >
                        <h3>
                          Leaders
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul
                          className={null}
                          id="two-cell"
                        >
                          <li
                            key="cleric"
                          >
                            <Link
                              replace={false}
                              to="//cleric"
                            >
                              <a
                                href="//cleric"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="cleric"
                                  onClick={[Function]}
                                >
                                  cleric
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="bard"
                          >
                            <Link
                              replace={false}
                              to="//bard"
                            >
                              <a
                                href="//bard"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="bard"
                                  onClick={[Function]}
                                >
                                  bard
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="2"
                      >
                        <h3>
                          Strikers
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul
                          className={null}
                          id={null}
                        >
                          <li
                            key="monk"
                          >
                            <Link
                              replace={false}
                              to="//monk"
                            >
                              <a
                                href="//monk"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="monk"
                                  onClick={[Function]}
                                >
                                  monk
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="ranger"
                          >
                            <Link
                              replace={false}
                              to="//ranger"
                            >
                              <a
                                href="//ranger"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="ranger"
                                  onClick={[Function]}
                                >
                                  ranger
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="rogue"
                          >
                            <Link
                              replace={false}
                              to="//rogue"
                            >
                              <a
                                href="//rogue"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="rogue"
                                  onClick={[Function]}
                                >
                                  rogue
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="warlock"
                          >
                            <Link
                              replace={false}
                              to="//warlock"
                            >
                              <a
                                href="//warlock"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="warlock"
                                  onClick={[Function]}
                                >
                                  warlock
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                      <div
                        key="3"
                      >
                        <h3>
                          Controllers
                        </h3>
                        <div
                          className="horizontal-line no-bottom"
                        />
                        <ul
                          className="odd-number"
                          id={null}
                        >
                          <li
                            key="druid"
                          >
                            <Link
                              replace={false}
                              to="//druid"
                            >
                              <a
                                href="//druid"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="druid"
                                  onClick={[Function]}
                                >
                                  druid
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="sorcerer"
                          >
                            <Link
                              replace={false}
                              to="//sorcerer"
                            >
                              <a
                                href="//sorcerer"
                                onClick={[Function]}
                              >
                                <button
                                  className={null}
                                  name="sorcerer"
                                  onClick={[Function]}
                                >
                                  sorcerer
                                </button>
                              </a>
                            </Link>
                          </li>
                          <li
                            key="wizard"
                          >
                            <Link
                              replace={false}
                              to="//wizard"
                            >
                              <a
                                href="//wizard"
                                onClick={[Function]}
                              >
                                <button
                                  className="double-size-btn"
                                  name="wizard"
                                  onClick={[Function]}
                                >
                                  wizard
                                </button>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </div>
                    </ul>
                  </Route>
                </Switch>
              </section>
            </MenuClass>
          </Connect(MenuClass)>
        </Route>
      </withRouter(Connect(MenuClass))>
    </Router>
  </MemoryRouter>
</Provider>
`;
